spec/requests/portfolio_items_spec.rb

```diff
@@ -200,4 +202,69 @@
       end
     end
   end
+
+  describe "copying portfolio items" do
+    let(:copy_portfolio_item) do
+      post "#{api("1.0")}/portfolio_items/#{portfolio_item.id}/copy", :params => params, :headers => default_headers
+    end
+
+    context "when copying into the same portfolio" do
+      let(:params) { { :portfolio_id => portfolio.id } }
+
+      before do
+        copy_portfolio_item
+      end
+
+      it "returns a 200" do
+        expect(response).to have_http_status(:ok)
+      end
+
+      it "returns a copy of the portfolio_item" do
+        expect(json["description"]).to eq portfolio_item.description
+        expect(json["owner"]).to eq portfolio_item.owner
+        expect(json["workflow_ref"]).to eq portfolio_item.workflow_ref
+      end
+
+      it "modifies the name to not collide" do
+        expect(json["name"]).to_not eq portfolio_item.name
```

Comment:
@lindgrenj6 What happens if this is called multiple times?

Reply:
Currently the names will collide. I'll add some logic in the service class to add a counter like you are suggesting.